src/tests/ftest/server/metadata.py

```diff
@@ -6,37 +6,39 @@
 """
 import traceback
 import uuid
-import threading
-import queue
 
 from avocado.core.exceptions import TestFail
 
 from apricot import TestWithServers
 from ior_utils import IorCommand
 from command_utils_base import CommandFailure
 from job_manager_utils import Orterun
+from thread_manager import ThreadManager
 
 
-def ior_runner_thread(manager, uuids, results):
-    """IOR run thread method.
+def run_ior_loop(manager, uuids):
+    """IOR run for each UUID provided.
 
     Args:
         manager (str): mpi job manager command
-        uuids (list): [description]
-        results (queue): queue for returning thread results
+        uuids (list): list of container UUIDs
+
+    Returns:
+        list: a list of CmdResults from each ior command run
+
     """
+    results = []
+    errors = []
     for index, cont_uuid in enumerate(uuids):
         manager.job.dfs_cont.update(cont_uuid, "ior.cont_uuid")
         try:
-            manager.run()
+            results.append(manager.run())
         except CommandFailure as error:
-            print(
-                "--- FAIL --- Thread-{0} Failed to run IOR {1}: "
-                "Exception {2}".format(
-                    index,
-                    "read" if "-r" in manager.job.flags.value else "write",
-                    str(error)))
-            results.put("FAIL")
+            errors.append("IOR Loop {}/{} failed: {}".format(index, len(uuids), error))
```

Comment:
What if any additional context could be included in the error string? Seems like the specific cont_uuid might be handy. If it could (like the original code does) include whether it is a write or read workload that would be handy. And not sure if it's possible to specify a thread index of some kind (not sure if it would have to be 0-9 in the example of 5 threads per read and write operation - or if threads could be identified as write 0-4 and read 0-4)?

Reply:
The `CommandFailure` (`error`) should provide the context of the ior command, but we could enhance the error message to include the detail.